src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/SuppressionEngine.cs

```diff
@@ -104,19 +104,22 @@ public bool IsErrorSuppressed(Suppression error)
         public void AddSuppression(Suppression suppression) => _suppressions.Add(suppression);
 
         /// <inheritdoc/>
-        public IReadOnlyCollection<Suppression> WriteSuppressionsToFile(string suppressionOutputFile, bool preserveUnnecessarySuppressions = false)
+        public (bool SuppressionFileUpdated, IReadOnlyCollection<Suppression> UpdatedSuppressions)
+            WriteSuppressionsToFile(string suppressionOutputFile, bool preserveUnnecessarySuppressions = false)
         {
             // If unnecessary suppressions should be preserved in the suppression file, union the
             // baseline suppressions with the set of actual suppressions. Duplicates are ignored.
-            HashSet<Suppression> suppressionsToSerialize = new(_suppressions);
+            HashSet<Suppression> suppressionsToSerialize = [.. _suppressions];
```

Comment:
I'm not so sure this is better syntax.  It doesn't call the constructor that takes a collection, so it initializes with default capacity then Add's N times.  This can result in reallocation and copies.  I think you did this because an analyzer told you to -- maybe we should also follow up with that analyzer.

Reply:
I opened https://github.com/dotnet/roslyn/issues/75981.

Reply:
Yeah, analyzer in this case.